nntrainer/utils/tracer.h

```diff
@@ -0,0 +1,222 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2022 Jiho Chu <jiho.chu@samsung.com>
+ *
+ * @file   tracer.h
+ * @date   23 December 2022
+ * @brief  Trace abstract class
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jiho Chu <jiho.chu@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+
+#ifndef __TRACER_H__
+#define __TRACER_H__
+
+#include <exception>
+#include <fstream>
+#include <list>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <tuple>
+#include <type_traits>
+
+namespace nntrainer {
```

Comment:
Can you hide the class Tracer if TRACER is not defined? (it will make the resulting binary even lighter)

Reply:
It's wrapped with ifdef.